app/middleware/gp.js

```diff
@@ -97,33 +97,11 @@ function upperCaseGpId(req, res, next) {
   next();
 }
 
-function getBookOnlineLink(req, res, next) {
+function getBookOnlineUrl(req, res, next) {
   const gpId = req.params.gpId;
-  const systemSupplier = cache.get(gpId).supplier_name;
 
-  switch (systemSupplier) {
-    case 'EMIS':
-      // eslint-disable-next-line no-param-reassign
-      req.gpDetails.bookOnlineLink = 'https://patient.emisaccess.co.uk/Account/Login';
-      break;
-    case 'Informatica':
-    case 'INPS':
-      // eslint-disable-next-line no-param-reassign
-      req.gpDetails.bookOnlineLink = 'https://www.myvisiononline.co.uk/vpp/';
-      break;
-    case 'Microtest':
-      // eslint-disable-next-line no-param-reassign
-      req.gpDetails.bookOnlineLink = 'https://www.thewaiting-room.net/';
-      break;
-    case 'TPP':
-      // eslint-disable-next-line no-param-reassign
-      req.gpDetails.bookOnlineLink = `https://systmonline.tpp-uk.com/Login?PracticeId=${gpId}`;
-      break;
-    case 'NK':
-    default:
-      // eslint-disable-next-line no-param-reassign
-      req.gpDetails.bookOnlineLink = '';
-  }
+  // eslint-disable-next-line no-param-reassign
+  req.gpDetails.bookOnlineUrl = cache.get(gpId).book_online_url;
```

Comment:
The URL should probably be validated here or, preferably, on load from the csv file.